flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/abilities/source/WatermarkPushDownSpec.java

```diff
@@ -75,15 +74,15 @@ public void apply(DynamicTableSource tableSource, SourceAbilityContext context)
         if (tableSource instanceof SupportsWatermarkPushDown) {
             GeneratedWatermarkGenerator generatedWatermarkGenerator =
                     WatermarkGeneratorCodeGenerator.generateWatermarkGenerator(
-                            context.getTableConfig().getConfiguration(),
+                            context.getTableConfig(),
                             context.getSourceRowType(),
                             watermarkExpr,
                             Option.apply("context"));
-            Configuration configuration = context.getTableConfig().getConfiguration();
 
             WatermarkGeneratorSupplier<RowData> supplier =
                     new GeneratedWatermarkGeneratorSupplier(
-                            configuration, generatedWatermarkGenerator);
+                            context.getTableConfig().getConfiguration(),
```

Comment:
still `getConfiguration` here?

Reply:
Currently no, because `WatermarkGenerator` implements `AbstrachRichFunction` for which the `open()` method has the following signature: 
```public void open(Configuration parameters) throws Exception {}```